src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -94,8 +97,23 @@ private static Path getSessionFilePath() {
 
     String run(Instant startInstant) throws Exception {
         Optional<Session> session = getCurrentSession();
-        if (session.isPresent() && sessionIsActive(startInstant, session.get()))
-            return session.get().profileName;
+        Instant expiry = startInstant.plus(864000 - 30, ChronoUnit.SECONDS);
```

Comment:
:bug: This is incorrect. You can't know this until you get it from AssumeRoleWithSAMLResult. Doing this here creates:
* at best: coupling of the expiry so it needs to be changed in multiple places
* at worst: significant and confusing expiry bugs

It's also Okta Admin console configuration and unknowable at compile time.